pkg/events/derive/hidden_kernel_module.go

```diff
@@ -38,34 +59,121 @@ func deriveHiddenKernelModulesArgs() deriveArgsFunction {
+
+		return []interface{}{addrHex, name, srcversion}, nil
 	}
 }
 
-func InitFoundHiddenModulesCache() error {
+// newModsCheckForHidden monitors only new added modules (added while tracee is running),
+// and reports if they are hidden
+func newModsCheckForHidden(startScanTime uint64, flags uint32) error {
+	// Since in old kernels it is not possible to iterate on a hashmap, the job is done here (userspace).
+	// It goes over a map that is maintained in the eBPF code (on each module insertion/deletion the map is updated),
+	// and checks if each module was seen (in modules list), and when was it inserted (to address race conditions).
+	// Based on this information, it determines whether it's hidden or not.
+	// If found hidden, it sends a message in the channel, which causes the address and the flags to get passed
+	// to the lkm submitter program (eBPF), which sends it back to userspace,
+	// this time with flags that will cause it to get submitted to the user as an event,
+	err := capabilities.GetInstance().Specific(
+		func() error {
+			var iter = newModuleOnlyMap.Iterator()
+			for iter.Next() {
+				addr := binary.LittleEndian.Uint64(iter.Key())
+				curVal, err := newModuleOnlyMap.GetValue(unsafe.Pointer(&addr))
+				if err != nil {
+					return err
+				}
+				insertTime := binary.LittleEndian.Uint64(curVal[0:8])
+				lastSeenTime := binary.LittleEndian.Uint64(curVal[8:16])
+				if insertTime <= startScanTime && lastSeenTime < startScanTime {
+					// It was inserted before the current scan, and we did not see it in the scan - it's hidden
+					// The receiving end will receive the message, trigger the lkm seeker submitter with a specific hidden module
+					// Note that we haven't really checked if the module is in the cache before,
+					// as we only have the address now.
+					if _, found := foundHiddenKernModsCache.Get(addr); !found {
+						wakeupChannel <- ScanRequest{Address: addr, Flags: flags} // It's hidden - and we haven't reported it before - report.
+					}
+				}
+			}
+
+			err := iter.Err()
+			if err != nil {
+				logger.Errorw("clearMap iterator received an error", "error", err.Error())
+				return iter.Err()
+			}
+
+			return nil
+		},
+		cap.SYS_ADMIN,
+	)
+
+	return err
+}
+
+// InitHiddenKernelModules initializes the module components
+func InitHiddenKernelModules(modsMap *bpf.BPFMap, newModMap *bpf.BPFMap, deletedModMap *bpf.BPFMap, insertedModMap *bpf.BPFMap) error {
+	allModsMap = modsMap
+	newModuleOnlyMap = newModMap
+	recentDeletedModulesMap = deletedModMap
+	recentInsertedModulesMap = insertedModMap
+
 	var err error
 	foundHiddenKernModsCache, err = lru.New(2048)
+	if err != nil {
+		logger.Errorw("Error occurred initializing kernel hidden modules: " + err.Error())
+	}
 	return err
 }
 
+// clearMap a utility to clear a map
 func clearMap(bpfMap *bpf.BPFMap) error {
-	err := capabilities.GetInstance().EBPF(
+	err := capabilities.GetInstance().Specific(
```

Comment:
Should use the `EBPF` capabilities RING (it already contains the base line for all capabilities needed by picked events PLUS the ones needed for eBPF functions).